_posts/2021-08-09-image-resizing.md

```diff
@@ -0,0 +1,172 @@
+        <img class="blog-image" alt="comparison on grid" src="/images/image-resizing/grid_comparison.png" >
+        <figcaption>Figure 2. Comparison of methods on the synthetic image of a grid.</figcaption>
+    </figure>
+</div>
+
+The previous considerations are valid also in this case, and it is clear how without the antialiasing the resulting images do not represent correctly the input.
+If you look at `OpenCV` results, it is curious that they are, in practice, the same results given by the _nearest neighbor_ filter of `Pillow`. It seems like that incorrectly implemented filters inadvertently resemble the naive approach.
+
+Other interesting results can be achieved using a pattern similar to the previous grid but with a border enlarged to 22 pixels and 20x20 white squares (note that we use a slightly different value for the border and the squares).
+
+<div class="blog-image-container">
+    <figure>
+        <img class="blog-image" alt="comparison on grid" src="/images/image-resizing/grid2_comparison.png" >
+        <figcaption>
+        Figure 3. Comparison of Pillow and OpenCV on a grid image.
+        From left to right, raw image (652x652), Pillow resized image with the nearest filter, OpenCV resized image with the nearest filter, Pillow resized image with the bilinear filter and OpenCV resized image with the bilinear filter.  
+        All the low-resolution images are 64x64 and are displayed with the same size as the raw image only for a better view.
+        </figcaption>
+    </figure>
+</div>
+
+With this input, the number of squares in the resulting images is unchanged, but in some cases, some rows or columns have a rectangular shape.
+
+### Natural images
+
+We chose to use synthetic images because we want to evidence the creation of artifacts and see the differences between the libraries. These differences are more difficult to see on natural images, i.e., the ones acquired with a camera. In this case, as you see in the figures below, the results with the _nearest neighbor_ filter are quite the same while, using the _bilinear_ one, the effect of the antialiasing filter becomes more visible.
+
+<div class="blog-image-container">
+    <figure>
+        <img class="blog-image" alt="nearest neighbor on dog" src="/images/image-resizing/dog_nearest.png" >
+        <figcaption>Figure 4. Nearest neighbor interpolation on a natural image.</figcaption>
+    </figure>
+</div>
+<div class="blog-image-container">
+    <figure>
+        <img class="blog-image" alt="bilinear on dog" src="/images/image-resizing/dog_bilinear.png" >
+        <figcaption>Figure 5. Bilinear interpolation on a natural image.</figcaption>
+    </figure>
+</div>
+
+<br/>
+
+## Conclusion
+
+In this post, we saw that even if image resizing is one of the most used image processing operations, it could hide some traps. It is essential to carefully choose the library we want to use to perform resize operations, particularly if you're going to deploy your ML solution.
```

Comment:
```suggestion
This article demonstrated that even if image resizing is one of the most used image processing operations, it could hide some traps. It is essential to carefully choose the library you want to use to perform resize operations, particularly if you're going to deploy your ML solution.
```